proposals/flexible-vectors/BinaryFlexibleVectors.md

```diff
@@ -0,0 +1,313 @@
+
+Non-saturating integer arithmetic:
+
+| Instruction              | `prefix` | `vecop` | Immediate operands       |
+| -------------------------|---------:|--------:|--------------------------|
+| `vec.i8.add`             |    `OxFA`|   `0x30`| -                        |
+| `vec.i16.add`            |    `0x79`|   `0x30`| -                        |
+| `vec.i32.add`            |    `0x78`|   `0x30`| -                        |
+| `vec.i64.add`            |    `0x77`|   `0x30`| -                        |
+| `vec.i8.sub`             |    `OxFA`|   `0x31`| -                        |
+| `vec.i16.sub`            |    `0x79`|   `0x31`| -                        |
+| `vec.i32.sub`            |    `0x78`|   `0x31`| -                        |
+| `vec.i64.sub`            |    `0x77`|   `0x31`| -                        |
+| `vec.i8.mul`             |    `OxFA`|   `0x32`| -                        |
+| `vec.i16.mul`            |    `0x79`|   `0x32`| -                        |
+| `vec.i32.mul`            |    `0x78`|   `0x32`| -                        |
+| `vec.i64.mul`            |    `0x77`|   `0x32`| -                        |
+| `vec.i8.neg`             |    `OxFA`|   `0x33`| -                        |
+| `vec.i16.neg`            |    `0x79`|   `0x33`| -                        |
+| `vec.i32.neg`            |    `0x78`|   `0x33`| -                        |
+| `vec.i64.neg`            |    `0x77`|   `0x33`| -                        |
+
+Saturating integer arithmetic:
+
+| Instruction              | `prefix` | `vecop` | Immediate operands       |
+| -------------------------|---------:|--------:|--------------------------|
+| `vec.i8.add_sat_u`       |    `OxFA`|   `0x40`| -                        |
+| `vec.i16.add_sat_u`      |    `0x79`|   `0x40`| -                        |
+| `vec.i32.add_sat_u`      |    `0x78`|   `0x40`| -                        |
+| `vec.i64.add_sat_u`      |    `0x77`|   `0x40`| -                        |
+| `vec.i8.add_sat_s`       |    `OxFA`|   `0x41`| -                        |
+| `vec.i16.add_sat_s`      |    `0x79`|   `0x41`| -                        |
+| `vec.i32.add_sat_s`      |    `0x78`|   `0x41`| -                        |
+| `vec.i64.add_sat_s`      |    `0x77`|   `0x41`| -                        |
+| `vec.i8.sub_sat_u`       |    `OxFA`|   `0x42`| -                        |
+| `vec.i16.sub_sat_u`      |    `0x79`|   `0x42`| -                        |
+| `vec.i32.sub_sat_u`      |    `0x78`|   `0x42`| -                        |
+| `vec.i64.sub_sat_u`      |    `0x77`|   `0x42`| -                        |
+| `vec.i8.sub_sat_s`       |    `OxFA`|   `0x43`| -                        |
+| `vec.i16.sub_sat_s`      |    `0x79`|   `0x43`| -                        |
+| `vec.i32.sub_sat_s`      |    `0x78`|   `0x43`| -                        |
+| `vec.i64.sub_sat_s`      |    `0x77`|   `0x43`| -                        |
+| `vec.i8.min_u`           |    `OxFA`|   `0x44`| -                        |
```

Comment:
is min saturating?

Reply:
Not really, this is because we have these operations under 'saturating' in SIMD. I think everything from here until the end of the section needs to move to regular arithmetic.

Reply:
Oh yes, https://github.com/WebAssembly/simd/blob/main/proposals/simd/SIMD.md I think that's incorrect, we should not repeat the mistake here :)

Reply:
Fixed